The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8079220146/job/22073160016) [(plain)](https://github.com/rust-lang/rust/commit/e4781115f2a5e50d60673d5c64d49182fc190247/checks/22073160016/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=Voultapher
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_b4716b23-9dba-4fd2-b860-e5e43e226d5f
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=a0d20eabb0d59fba473f4d2e0a0ab4249c23a5f4
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_b4716b23-9dba-4fd2-b860-e5e43e226d5f
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_b4716b23-9dba-4fd2-b860-e5e43e226d5f
GITHUB_TRIGGERING_ACTOR=Voultapher
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/121732/merge
GITHUB_WORKFLOW_SHA=a0d20eabb0d59fba473f4d2e0a0ab4249c23a5f4
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
#13 3.538 Building wheels for collected packages: reuse
#13 3.539   Building wheel for reuse (pyproject.toml): started
#13 3.863   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.864   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.864   Stored in directory: /tmp/pip-ephem-wheel-cache-prgrq3_a/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.867 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.888   Attempting uninstall: setuptools
#13 3.889     Found existing installation: setuptools 59.6.0
#13 3.890     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
   Compiling compiler_builtins v0.1.108
error: unresolved link to `matches`
   --> library/core/src/macros/mod.rs:121:78
    |
121 | /// The pattern syntax is exactly the same as found in a match arm and the [`matches!`] macro. The
    |
    |
    = note: `macro_rules` named `matches` exists in this crate, but it is not in scope at this link's location
    = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(rustdoc::broken_intra_doc_links)]`
error: unresolved link to `matches`
   --> library/core/src/macros/mod.rs:388:78
    |
    |
388 | /// The pattern syntax is exactly the same as found in a match arm and the [`matches!`] macro. The
    |
    |
    = note: `macro_rules` named `matches` exists in this crate, but it is not in scope at this link's location
error: could not document `core`

Caused by:
Caused by:
  process didn't exit successfully: `/checkout/obj/build/bootstrap/debug/rustdoc --edition=2021 --crate-type lib --crate-name core library/core/src/lib.rs --target x86_64-unknown-linux-gnu -o /checkout/obj/build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/doc/x86_64-unknown-linux-gnu/doc -Zunstable-options --check-cfg 'cfg(feature, values("debug_refcell", "panic_immediate_abort"))' --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat -C metadata=a1aec6da431e1dc8 -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/doc/x86_64-unknown-linux-gnu/release/deps -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/doc/release/deps --cfg=bootstrap -Csymbol-mangling-version=legacy -Zunstable-options '--check-cfg=cfg(feature,values(any()))' -Zunstable-options '--check-cfg=cfg(bootstrap)' '--check-cfg=cfg(stdarch_intel_sde)' '--check-cfg=cfg(no_fp_fmt_parse)' '--check-cfg=cfg(no_global_oom_handling)' '--check-cfg=cfg(no_rc)' '--check-cfg=cfg(no_sync)' '--check-cfg=cfg(netbsd10)' '--check-cfg=cfg(backtrace_in_libstd)' '--check-cfg=cfg(target_env,values("libnx","p2"))' '--check-cfg=cfg(target_arch,values("spirv","nvptx","xtensa"))' --document-private-items --document-hidden-items -Dwarnings '-Wrustdoc::invalid_codeblock_attributes' --crate-version '1.78.0-nightly (a0d20eabb 2024-02-28)' '-Zcrate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/")' '-Zcrate-attr=warn(rust_2018_idioms)' -Z unstable-options --resource-suffix 1.78.0 --markdown-css rust.css --markdown-no-toc --index-page /checkout/src/doc/index.md` (exit status: 1)
Build completed unsuccessfully in 0:00:08
  local time: Wed Feb 28 11:10:31 UTC 2024
  network time: Wed, 28 Feb 2024 11:10:31 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/library/core/src/macros/mod.rs
+++ b/library/core/src/macros/mod.rs
@@ -118,7 +118,7 @@ macro_rules! assert_ne {
 /// the debug representation, of the actual value shape that did not meet expectation. In contrast
 /// using [`assert!`] will only print that the expectation was not met, but not why.
 ///
-/// The pattern syntax is exactly the same as found in a match arm and the [`matches!`] macro. The
+/// The pattern syntax is exactly the same as found in a match arm and the `matches!` macro. The
 /// optional if guard can be used to add additional checks that must be true for the matched value,
 /// otherwise this macro will panic.
 ///
@@ -385,7 +385,7 @@ macro_rules! debug_assert_ne {
 /// print the debug representation, of the actual value shape that did not meet expectation. In
 /// contrast using [`debug_assert!`] will only print that the expectation was not met, but not why.
 ///
-/// The pattern syntax is exactly the same as found in a match arm and the [`matches!`] macro. The
+/// The pattern syntax is exactly the same as found in a match arm and the `matches!` macro. The
 /// optional if guard can be used to add additional checks that must be true for the matched value,
 /// otherwise this macro will panic.
 ///
@@ -430,10 +430,15 @@ pub macro debug_assert_matches($($arg:tt)*) {
     }
 }
 
-/// Returns whether the given expression matches any of the given patterns.
+/// Returns whether the given expression matches the provided pattern.
 ///
-/// Like in a `match` expression, the pattern can be optionally followed by `if`
-/// and a guard expression that has access to names bound by the pattern.
+/// The pattern syntax is exactly the same as found in a match arm. The optional if guard can be
+/// used to add additional checks that must be true for the matched value, otherwise this macro will
+/// return `false`.
+///
+/// When testing that a value matches a pattern, it's generally preferable to use
+/// [`assert_matches!`] as it will print the debug representation of the value if the assertion
+/// fails.
 ///
 /// # Examples
 ///

--- a/library/core/src/macros/mod.rs
+++ b/library/core/src/macros/mod.rs
@@ -115,8 +115,8 @@ macro_rules! assert_ne {
 /// Asserts that an expression matches the provided pattern.
 ///
 /// This macro is generally preferable to `assert!(matches!(value, pattern))`, because it can print
-/// the debug representation, of the actual value shape that did not meet expectation. In contrast,
-/// using [`assert!`] will only print that the expectation was not met, but not why.
+/// the debug representation of the actual value shape that did not meet expectations. In contrast,
+/// using [`assert!`] will only print that expectations were not met, but not why.
 ///
 /// The pattern syntax is exactly the same as found in a match arm and the `matches!` macro. The
 /// optional if guard can be used to add additional checks that must be true for the matched value,
@@ -382,8 +382,8 @@ macro_rules! debug_assert_ne {
 /// Asserts that an expression matches the provided pattern.
 ///
 /// This macro is generally preferable to `debug_assert!(matches!(value, pattern))`, because it can
-/// print the debug representation, of the actual value shape that did not meet expectation. In
-/// contrast, using [`debug_assert!`] will only print that the expectation was not met, but not why.
+/// print the debug representation of the actual value shape that did not meet expectations. In
+/// contrast, using [`debug_assert!`] will only print that expectations were not met, but not why.
 ///
 /// The pattern syntax is exactly the same as found in a match arm and the `matches!` macro. The
 /// optional if guard can be used to add additional checks that must be true for the matched value,

--- a/library/core/src/macros/mod.rs
+++ b/library/core/src/macros/mod.rs
@@ -115,7 +115,7 @@ macro_rules! assert_ne {
 /// Asserts that an expression matches the provided pattern.
 ///
 /// This macro is generally preferable to `assert!(matches!(value, pattern))`, because it can print
-/// the debug representation, of the actual value shape that did not meet expectation. In contrast
+/// the debug representation, of the actual value shape that did not meet expectation. In contrast,
 /// using [`assert!`] will only print that the expectation was not met, but not why.
 ///
 /// The pattern syntax is exactly the same as found in a match arm and the `matches!` macro. The
@@ -383,7 +383,7 @@ macro_rules! debug_assert_ne {
 ///
 /// This macro is generally preferable to `debug_assert!(matches!(value, pattern))`, because it can
 /// print the debug representation, of the actual value shape that did not meet expectation. In
-/// contrast using [`debug_assert!`] will only print that the expectation was not met, but not why.
+/// contrast, using [`debug_assert!`] will only print that the expectation was not met, but not why.
 ///
 /// The pattern syntax is exactly the same as found in a match arm and the `matches!` macro. The
 /// optional if guard can be used to add additional checks that must be true for the matched value,

